src/runtime/metal.cpp

```diff
@@ -542,10 +555,11 @@ WEAK int halide_metal_initialize_kernels(void *user_context, void **state_ptr, c
     mtl_library *library{};
     if (!compilation_cache.kernel_state_setup(user_context, state_ptr, metal_context.device, library,
                                               new_library_with_source, metal_context.device,
-                                              source, source_size)) {
+                                              source, source_size) ||
+        library == nullptr) {
```

Comment:
Nit: indentation

Reply:
This what clang-format did, let me see if I can pacify it otherwise